src/motherduck_destination_server.cpp

```diff
@@ -364,9 +368,19 @@ DestinationSdkImpl::Test(::grpc::ServerContext *context,
         find_property(request->configuration(), MD_PROP_DATABASE);
     std::unique_ptr<duckdb::Connection> con =
         get_connection(request->configuration(), db_name);
-    check_connection(*con);
+
+    if (request->name() == CONFIG_TEST_NAME_AUTHENTICATE) {
+      check_connection(*con);
+      response->set_success(true);
+    } else {
+      auto const err = "Unknown test requested: <" + request->name() + ">";
+      mdlog::severe(err);
+      return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT, err);
+    }
+    response->set_success(true);
   } catch (const std::exception &e) {
     mdlog::severe("Test endpoint failed: " + std::string(e.what()));
+    response->set_success(false);
     response->set_failure(e.what());
```

Comment:
```suggestion
    response->set_failure("Data source configuration test for source " + db_name + " failed: " + std::string(e.what()));
```
So this for running tests against a data source, is that right? And typically a user interacts with this via the web? I', wondering, what kind of error message does the user see if it fails? Does it look reasonable or confusing? Based on the code I'd say it will return something like this:
```
<some data source-specific authentication error message>
```

Would it make sense if it would clarify the top-level error as well?
```
Data source configuration test for source <source name> failed: <some data source-specific authentication error message>
```

No need to add this if this would make it too verbose though

Reply:
yeah, I agree -- the more information, the better. Especially since we don't actually have easy access to server logs, the more information user-facing messages provide, the better. This is what (a different) error looks like on fivetran:
![image](https://github.com/MotherDuck-Open-Source/motherduck-fivetran-connector/assets/41136058/06b1c8f4-9f67-4345-8164-9d35963524e9)

This is what a config check for a different integration (google sheets) looks like:
![image](https://github.com/MotherDuck-Open-Source/motherduck-fivetran-connector/assets/41136058/38f7fcfa-1208-4661-bb59-172607b95faf)

Reply:
ah, even in testing it's much better:
```
 REQUIRE( response.failure() == "Missing property motherduck_token" )
with expansion:
  "Authentication test for database <fivetran_test> failed: Missing property
  motherduck_token"
  ==
  "Missing property motherduck_token"
```

Although I had to split the `try/catch` block into two blocks -- the database name retrieval can also fail.